EmbeddedPkg/Library/NonCoherentDmaLib/NonCoherentDmaLib.c

```diff
@@ -654,7 +656,7 @@ DmaFreeBuffer (
     }
   }
 
-  if (!Found) {
+  if (!Found || (Alloc == NULL)) {

```

Comment:
I don't understand this change.
Alloc is dereferenced on every pass through the loop, so why would we check after the loop if it's NULL?

Presumably this is a false positive in the tool?

If you can rewrite this in a way that makes sense *and* makes the tool happy, I'm happy to take that change, but not a patch that adds confusion without solving actual problems.

Reply:
This appears to be a false positive from the tool. Sorry about that.